### PR TITLE
feat: Update used aliquots data when primary aliquot is updated for a library

### PR DESCRIPTION
**Pull request by stevieing**
Closes #

#### Changes proposed in this pull request

- 

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: app/models/pacbio/library.rb [hunk 1]
```diff
@@ -49,6 +49,8 @@ class Library < ApplicationRecord
     # if the new volume is greater than the used volume.
     before_update :primary_aliquot_volume_sufficient
 
+    after_update :update_used_aliquots
```
Comment (BenTopping) [inline app/models/pacbio/library.rb:52]: Will this run before this https://github.com/sanger/traction-service/blob/develop/app/resources/v1/pacbio/library_resource.rb#L39? Could we add a test for it as this resource method should update the warehouse with the updated tag data.

**Review by BenTopping — COMMENTED** [review]
Looks good just one comment about potential race condition.
Do we also need a script to fix the prod data?

File: app/models/pacbio/library.rb (hunk 1 above)
Comment (stevieing) [reply app/models/pacbio/library.rb:52]: Not sure if this is easily testable but "in Ruby on Rails, callbacks such as after_update run within the transaction of the update action. This means if you have an after_update callback that makes changes to the database and an error occurs after the update action but within the same transaction, all changes including those made in the after_update callback will be rolled back. This ensures data integrity by not applying partial updates to the database."

We can test on UAT and training.

Comment (BenTopping) [reply app/models/pacbio/library.rb:52]: Sure, yeah easiest way will be to try on UAT. You might be able to manually test it by putting a debugger in the resource after_update method and checking if the data there has been updated.

**Review by BenTopping — APPROVED** [review]

**Pull request merged**